imagemounter/dependencies.py

```diff
@@ -0,0 +1,153 @@
+from imagemounter import _util
+
+
+class Dependency(object):
+    installed_explanation = "INSTALLED {!s}"
+
+    def __init__(self, name, package="", why=""):
+        self.name = name
+        self.package = package
+        self.why = why
+
+    @property
+    def status(self):
+        if self.check():
+            return self.installed_explanation.format(self)
+        else:
+            return "MISSING   {!s:<20}".format(self) + self.missing_explanation
+
+
+class CommandDependency(Dependency):
+
+    def __str__(self):
+        return self.name
+
+    def check(self):
+        """Check if the command is available on the system"""
+        return _util.command_exists(self.name)
```

Comment:
I think this should be a property named ``is_available`` or something, making it a simple:

```
if ewfmount.is_available:
    do_something_with_ewfmount()
else:
    raise DependencyError(ewfmount.missing_explanation)
```

Could also alias it in ``__bool__`` so you could just do ``if ewfmount:``, but not sure that's the best Pythonic way. How do you feel?

Reply:
I like calling it `is_available`, but I'm on the fence of whether it makes more sense as a property or a function. The advice I've heard is that properties shouldn't make "expensive" operations (like accessing a database) look "cheap" (like normal Python attribute access). Calling `command_exists` is sort of in the middle, but I'll go ahead and make it a property. 

Aliasing `__bool__` would definitely be interesting (it would allow you to write things like `if not all(ewfmount, xmount, affuse): ...`. But that may end up coming across as a bit **too** magic, so I'll leave it out for now, but wouldn't be opposed to adding it.

Reply:
I'm also not entirely certain whether a property is the way to go here. Perhaps making it a function is the way to go instead.